src/allotropy/exceptions.py

```diff
@@ -1,2 +1,15 @@
+from collections.abc import Collection
+from typing import Any, Optional
+
+
 class AllotropeConversionError(Exception):
     pass
+
+
+def msg_for_error_on_unrecognized_value(
+    key: str, value: Any, valid_values: Optional[Collection[Any]] = None
+) -> str:
+    msg = f"Unrecognized {key}: '{value}'."
+    if valid_values:
+        msg += f" Only {sorted(valid_values)} are supported."
```

Comment:
Hey @mdlu , this will fail with `TypeError` for types that can't be sorted. Could you please follow up to restrict valid_values' type?